python/ray/ml/config.py

```diff
@@ -35,29 +34,71 @@ class ScalingConfigDataClass:
         Strategies <pgroup-strategy>` for the possible options.
     """
 
-    num_workers: int
+    num_workers: int = 1
     use_gpu: bool = False
     resources_per_worker: Optional[Dict] = None
     placement_strategy: str = "PACK"
 
+    def __post_init__(self):
+        self.resources_per_worker = (
+            self.resources_per_worker if self.resources_per_worker else {}
+        )
+
+        if self.num_workers < 0:
+            raise ValueError("`num_workers` must be a positive integer.")
```

Comment:
This may need to be overridden by individual Trainers, but based on the current condition:
```suggestion
            raise ValueError("`num_workers` must be a nonnegative integer.")
```

Reply:
This has been removed, so no longer applicable!